src/database/engine/rrdengine.c

```diff
@@ -1688,6 +1762,75 @@ static inline void worker_dispatch_query_prep(struct rrdeng_cmd cmd, bool from_w
         work_dispatch(ctx, pdc, NULL, cmd.opcode, query_prep_tp_worker, NULL);
 }
 
+uint64_t get_directory_free_bytes_space(struct rrdengine_instance *ctx)
+{
+    uint64_t free_bytes = 0;
+    struct statvfs buff_statvfs;
+    if (statvfs(ctx->config.dbfiles_path, &buff_statvfs) == 0)
+        free_bytes = buff_statvfs.f_bavail * buff_statvfs.f_bsize;
+
+    return (free_bytes - (free_bytes * 5 / 100));
+}
+
+void dbengine_retention_statistics(void)
+{
+    static bool init = false;
+    static DBENGINE_TIER_STATS stats[RRD_STORAGE_TIERS];
+
+    for (size_t tier = 0; tier < storage_tiers; tier++) {
+        if (init == false) {
+            char id[200];
+            snprintfz(id, sizeof(id) - 1, "dbengine_retention_tier%zu", tier);
+            stats[tier].st = rrdset_create_localhost(
+                "netdata",
+                id,
+                NULL,
+                "dbengine",
+                "netdata.dbengine_tier_retention",
+                "dbengine space and time retention",
+                "%",
+                "netdata",
+                "stats",
+                200000,
+                10,
+                RRDSET_TYPE_LINE);
+
+            stats[tier].rd_space = rrddim_add(stats[tier].st, "space", NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+            stats[tier].rd_time = rrddim_add(stats[tier].st, "time", NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+
+            char tier_str[5];
+            snprintfz(tier_str, 4, "%ld", tier);
+            rrdlabels_add(stats[tier].st->rrdlabels, "dbengine_tier", tier_str, RRDLABEL_SRC_AUTO);
```

Comment:
```suggestion
            rrdlabels_add(stats[tier].st->rrdlabels, "tier", tier_str, RRDLABEL_SRC_AUTO);
```


I know that I suggested "dbengine_tier"...changing to just "tier" 😄